src/azure-cli/azure/cli/command_modules/containerapp/_utils.py

```diff
@@ -948,13 +948,13 @@ def _remove_readonly_attributes(containerapp_def):
             del containerapp_def['properties'][unneeded_property]
 
 
-# Remove null/None properties in a model since the PATCH API will delete those. Not needed once we move to the SDK
+# Remove null/None/empty properties in a model since the PATCH API will delete those. Not needed once we move to the SDK
 def clean_null_values(d):
     if isinstance(d, dict):
         return {
             k: v
             for k, v in ((k, clean_null_values(v)) for k, v in d.items())
-            if v or isinstance(v, list)
+            if (isinstance(v, dict) and len(v.items()) > 0) or (not isinstance(v, dict) and v is not None) or isinstance(v, list)
```

Comment:
which is v when `not isinstance(v, dict) and v is not None`

Reply:
In the old version of clean_null_values, it will remove {} too. I added a test to make sure they are consistent in `test_containerapp_clean_null_values.py`.